core/chain-configs/src/genesis_config.rs

```diff
@@ -235,6 +236,16 @@ impl Genesis {
         )
         .expect("Failed to create / write a genesis config file.");
     }
+
+    /// Hash of the json-serialized input.
+    /// DEVNOTE: the representation is not unique, and could change on upgrade.
+    pub fn json_hash(&self) -> CryptoHash {
```

Comment:
Why do we serialize it in json instead of borsh? Also it seems that we don't need to hash `Genesis` entirely since records will go into state root and it suffices to hash genesis config.

Reply:
This change uses the hash to avoid recomputing state roots. No changes to the protocol.

Genesis uses too many types that don't support borsh and some aren't in our crates, need a big refactor.